pkg/providers/vsphere/virtualmachine/configspec.go

```diff
@@ -51,6 +51,16 @@ func CreateConfigSpec(
 		},
 	)
 
+	// Ensure ExtraConfig contains the VM Class's reservation profile ID if set.
+	if id := vmClassSpec.ReservedProfileID; id != "" {
```

Comment:
How will we be using ReservedSlots in the VM configSpec?

Reply:
We are not. It's there for future use so I can report on how many free slots there are in the VM Class status. I just have not done that yet. But since Michal is going to push down the number of slots with the changes he is making, I wanted to give him a place to store it.